FrontEnd/src/components/Navigation.js

```diff
@@ -8,71 +8,75 @@ import { Row, Col } from 'react-materialize';
 import componentCSS from './Navigation.scss';
 
 class Navigation extends React.Component {
-	constructor(props) {
-		super(props);
+    constructor(props) {
+        super(props);
+        this.state = {
 
-		this.state = {
-			pages: [
-				{
-					icon: 'book',
-					text: 'Library',
-					ref: '/index.html',
-					class: 'valign-wrapper',
-				},
-				{
-					icon: 'settings',
-					text: 'Settings',
-					ref: '/settings.html',
-					class: 'valign-wrapper',
-				},
-				{
-					icon: 'face',
-					text: 'Admin',
-					ref: '/admin.html',
-					class: 'valign-wrapper',
-				},
-			],
-		};
+            pages: [
+                {
+                    icon: 'book',
+                    text: 'Library',
+                    class: 'valign-wrapper active',
+					isSelected: true,
+					callbackFn: () => {this.switchSelected(this.state.pages[0].text);},
+                },
+                {
+                    icon: 'settings',
+                    text: 'Settings',
+                    class: 'valign-wrapper',
+					isSelected: false,
+					callbackFn: () => {this.switchSelected(this.state.pages[1].text);},
+                },
+                {
+                    icon: 'face',
+                    text: 'Admin',
+                    class: 'valign-wrapper',
+					isSelected: false,
+					callbackFn: () => {this.switchSelected(this.state.pages[2].text);},
+                },
+            ],
 
-		switch (this.props.isActive) {
-			case 'Library':
-				this.state.pages[0].class += ' active';
-				break;
+        };
+    }
 
-			case 'Settings':
-				this.state.pages[1].class += ' active';
-				break;
+    render() {
+        return (
+            <div className={'navigation'}>
+                <Row>
+                    <Col s={12}>
+                        <ul className="mt0 mb0">
+                            {this.state.pages.map((navList) => (
```

Comment:
minor thing, but this is not really a 'list' but an 'object' or 'element'

Reply:
done this way as that's how I've typically seen navigation done but i can update it to a div?